app/src/main/java/com/mealsmadeeasy/data/FakeMealStore.kt

```diff
@@ -91,4 +92,64 @@ class FakeMealStore : MealStore {
             ?: Observable.just(emptyList<Ingredient>())!!
 
 
+    override fun addMealToMealPlan(meal: Meal, date: DateTime, mealPeriod: MealPeriod) {
+        var added = false
+        var changedPlan = mealPlan.value
+        mealPlan.value.meals
+                .filter { it.date.dayOfMonth() == date.dayOfMonth() && it.mealPeriod == mealPeriod }
+                .forEach {
+                    var temp = emptyList<MealPlanEntry>()
+                    mealPlan.value.meals.forEach {
+                        if (it.date.dayOfMonth() == date.dayOfMonth()
+                                && it.mealPeriod == mealPeriod) {
+                            temp += it.copy(meals = it.meals + meal)
+                        } else {
+                            temp += it
+                        }
+                    }
+                    changedPlan = mealPlan.value.copy(meals = temp)
+                    added = true
+                }
+
+        if (!added) {
```

Comment:
I think this condition could be improved because it's not immediately obvious when this happens. I'm guessing that it's when a meal plan entry doesn't exist for this day and meal, but I can't really tell that just by reading this method.